From 8cd32a57f7677a49cadbacb1a1a7ce9008ebfde6 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Wed, 6 Jun 2018 14:00:14 +0300
Subject: [PATCH 3920/5242] MLK-18534-2: ARM64: dts: mx8mm-evk: ak4497: use
 1:1 bclk:mclk ratio for DSD512

commit  7c33fd9b46805a80bd44c292717d2cd37ac1bccf from
https://source.codeaurora.org/external/imx/linux-imx.git

Since IP version 3.01 (845s) SAI has support for 1:1 bclk:mclk ratio.
Given this the specific DSD512 pinctrl option can be removed and SAI mclk
frequency decreased.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts   |   22 ++------------------
 1 file changed, 2 insertions(+), 20 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts
index 3e07f54..95b09f2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts
@@ -59,35 +59,17 @@
 				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
 			>;
 		};
-
-		pinctrl_sai1_dsd512: sai1grp_dsd512 {
-			fsl,pins = <
-				MX8MM_IOMUXC_SAI1_MCLK_SAI1_TX_BCLK	0xd6
-				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
-				MX8MM_IOMUXC_SAI1_RXD7_SAI1_TX_DATA4	0xd6
-				MX8MM_IOMUXC_SAI1_TXC_SAI1_TX_BCLK	0xd6
-				MX8MM_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0	0xd6
-				MX8MM_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1	0xd6
-				MX8MM_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2	0xd6
-				MX8MM_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3	0xd6
-				MX8MM_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4	0xd6
-				MX8MM_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5	0xd6
-				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
-				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
-			>;
-		};
 	};
 };
 
 &sai1 {
-	pinctrl-names = "default", "dsd", "dsd512";
+	pinctrl-names = "default", "dsd";
 	pinctrl-0 = <&pinctrl_sai1_pcm>;
 	pinctrl-1 = <&pinctrl_sai1_dsd>;
-	pinctrl-2 = <&pinctrl_sai1_dsd512>;
 	assigned-clocks = <&clk IMX8MM_CLK_SAI1_SRC>,
 			<&clk IMX8MM_CLK_SAI1_DIV>;
 	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL2_OUT>;
-	assigned-clock-rates = <0>, <45158400>;
+	assigned-clock-rates = <0>, <22579200>;
 	fsl,sai-multi-lane;
 	fsl,dataline,dsd = <0xff 0x11>;
 	dmas = <&sdma2 0 26 0>, <&sdma2 1 26 0>;
-- 
1.7.9.5

